libqtile/backend/wayland/core.py

```diff
@@ -981,8 +1104,16 @@ def _under_pointer(self) -> tuple[window.WindowType, Surface | None, float, floa
                             return win, None, 0, 0
         return None
 
-    def stack_windows(self) -> None:
-        """Put all windows of all types in a Z-ordered list."""
+    def stack_windows(self, restack: window.WindowType | None = None) -> None:
+        """
+        Put all windows of all types in a Z-ordered list.
+
+        A (non-layer_shell) window passed as 'restack' will bubble up the Z order.
+        """
+        if restack:
+            self.mapped_windows.remove(restack)
```

Comment:
Will `restack` always be in `self.mapped_windows`? There's no error handling here.

Reply:
Yes it always will be. It's not particularly pretty code, this stuff with `mapped_windows`. I'm hoping to rewrite some of this in the not so distant future (using some nice helpers from wlroots called wlr_scene that should handle this for us to some extent)